src/webln/NostrWeblnProvider.ts

```diff
@@ -644,6 +689,139 @@ export class NostrWebLNProvider implements WebLNProvider, Nip07Provider {
+          JSON.stringify(command),
+        );
+        const unsignedEvent: UnsignedEvent = {
+          kind: 23194,
+          created_at: Math.floor(Date.now() / 1000),
+          tags: [["p", this.walletPubkey]],
+          content: encryptedCommand,
+          pubkey: this.publicKey,
+        };
+
+        const event = await this.signEvent(unsignedEvent);
+        // subscribe to NIP_47_SUCCESS_RESPONSE_KIND and NIP_47_ERROR_RESPONSE_KIND
+        // that reference the request event (NIP_47_REQUEST_KIND)
+        const sub = this.relay.sub([
+          {
+            kinds: [23195],
+            authors: [this.walletPubkey],
+            "#e": [event.id],
+          },
+        ]);
+
+        function replyTimeout() {
+          sub.unsub();
+          //console.error(`Reply timeout: event ${event.id} `);
+          reject({
+            error: `reply timeout: event ${event.id}`,
+            code: "INTERNAL",
+          });
+        }
+
+        const replyTimeoutCheck = setTimeout(replyTimeout, 60000);
+
+        sub.on("event", async (event) => {
+          // console.log(`Received reply event: `, event);
+
+          const decryptedContent = await this.decrypt(
+            this.walletPubkey,
+            event.content,
+          );
+          // console.log(`Decrypted content: `, decryptedContent);
+          let response;
+          try {
+            response = JSON.parse(decryptedContent);
+          } catch (e) {
+            console.error(e);
+            clearTimeout(replyTimeoutCheck);
+            sub.unsub();
+            reject({ error: "invalid response", code: "INTERNAL" });
+            return;
+          }
+          if (event.kind == 23195 && response.result) {
+            // console.info("NIP-47 result", response.result);
+            try {
+              if (!resultValidator(response.result)) {
+                throw new Error(
+                  "Response from NWC failed validation: " +
+                    JSON.stringify(response.result),
+                );
+              }
+              const dTag = event.tags.find((tag) => tag[0] === "d")?.[1];
+              if (dTag === undefined) {
+                throw new Error("No d tag found in response event");
+              }
+              // console.info("dTag", dTag);
+              results.push({
+                ...response.result,
+                dTag,
+              });
+              if (results.length === numPayments) {
+                clearTimeout(replyTimeoutCheck);
+                sub.unsub();
+                //console.log("Received results", results);
+                resolve(results.map(resultMapper));
+                this.notify(weblnMethod, response.result);
+              }
+            } catch (error) {
+              console.error(error);
+              clearTimeout(replyTimeoutCheck);
+              sub.unsub();
+              reject({
+                error: (error as Error).message,
+                code: "INTERNAL",
+              });
```

Comment:
But do we want to stop the entire execution and reject it because one payment failed? I don't think we should as the other payments are processed by the NWC server anyways and that information would be lost.

Reply:
Update: we should do this, but we have other stuff to work on. There is already a TODO for now.